src/main/java/com/segment/analytics/android/integrations/nielsendcr/NielsenDCRIntegration.java

```diff
@@ -358,6 +368,57 @@ private JSONObject mapSpecialKeys(
     return adContentMetadata;
   }
 
+  public String formatAirdate(String airdate) {
+    String finalDate = airdate;
+
+    // assuming 'airdate' was passed as ISO date string per Segment spec
+    try {
+      Matcher s = shortDate.matcher(finalDate);
+      Matcher l = longDate.matcher(finalDate);
+
+      if (s.find()) {
+        finalDate =
+            new StringBuilder() //
```

Comment:
This is fine but a little verbose. I think you can also do:

`String.format(%s%s%s 00:00:00, s.group(1), s.group(2), s.group(3));`

but you should test to be sure if you want to bother changing it.